packages/generator/src/batch/imports.ts

```diff
@@ -36,5 +40,15 @@ export function importBatchDeclarations(
 }
 
 function getNamedImports(service: VdmServiceMetadata): string[] {
-  return service.entities.map(e => e.className);
+  const complexTypes = service.functionImports
+    .filter(
+      f =>
+        f.returnType.returnTypeCategory === VdmReturnTypeCategory.COMPLEX_TYPE
```

Comment:
```suggestion
      ({returnType}) =>
        returnType.returnTypeCategory === VdmReturnTypeCategory.COMPLEX_TYPE
```
[pp] Destructuring the parameter remove the need for the arbitrarily named `f`

Reply:
ah, I have to remember deconstructing also in array function